modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -425,17 +424,19 @@ public String getSchemaType(Schema p) {
         String ref = p.get$ref();
         String type;
 
-        if (ref != null && !ref.isEmpty()) {
+        if (ref != null && !ref.isEmpty() && ref.endsWith(openAPIType)) {
```

Comment:
from what i understand, this fix only works if the schema name happens to be `openAPIType` (e.g. string), right?

Reply:
If I recall correctly, `openAPIType` would be a primitive type name, (e.g., `"integer"`), for the reference referring to an integer, like [this example below](https://github.com/OpenAPITools/openapi-generator/pull/18785/files#diff-1d7a7292e42455e861837923ad2c7c6a12ddcd89f502e3813e7fb4b84164e774R2352).

And `openAPIType` would be the schema name (e.g., `"SomeSchema"`) for the reference to a schema object.

So, checking whether the `ref` is ending with the `openAPIType` can determine whether it's a Schema object, which needs `toModelName()` to capitalize its first letter, rather than a primitive type, which will be handled elsewhere.